flocker/node/_docker.py

```diff
@@ -543,14 +543,26 @@ def _list():
                     # have been removed in another thread.
                     if e.response.status_code == NOT_FOUND:
                         continue
+                    else:
+                        raise
 
                 state = (u"active" if data[u"State"][u"Running"]
                          else u"inactive")
                 name = data[u"Name"]
                 image = data[u"Config"][u"Image"]
                 command = data[u"Config"][u"Cmd"]
-                image_command = self._client.inspect_image(image)
-                if image_command[u"Config"][u"Cmd"] == command:
+                try:
+                    image_data = self._client.inspect_image(image)
+                except APIError as e:
+                    if e.response.status_code == NOT_FOUND:
+                        # Image has been deleted, so just fill in some
+                        # stub data so we can return *something*. This
+                        # should happen only for stopped containers so
+                        # some inaccuracy is acceptable.
+                        image_data = {u"Config": {u"Env": [], u"Cmd": []}}
```

Comment:
I observed that this can happen even for running containers.  Docker lets you untag an image that is being used by a running container:

```
(flocker)jean-paul@dilaton:~$ docker run -d busybox:latest sleep 60000
097f8f3f629b1b626e3482264eaf179c833f4e5dda95924610c43119f20f8ed7
(flocker)jean-paul@dilaton:~$ docker inspect -f "{{.Config.Image}}" 097f8f3f629b1b626e3482264eaf179c833f4e5dda95924610c43119f20f8ed7
busybox:latest
(flocker)jean-paul@dilaton:~$ docker rmi busybox:latest
Error response from daemon: Conflict, cannot delete 4986bf8c1536 because the running container 097f8f3f629b is using it, stop it and use -f to force
FATA[0000] Error: failed to remove one or more images   
(flocker)jean-paul@dilaton:~$ docker tag busybox:latest busybox:1.2.3.4
(flocker)jean-paul@dilaton:~$ docker rmi busybox:latest
Untagged: busybox:latest
(flocker)jean-paul@dilaton:~$ docker inspect -f "{{.Config.Image}}" 097f8f3f629b1b626e3482264eaf179c833f4e5dda95924610c43119f20f8ed7
busybox:latest
(flocker)jean-paul@dilaton:~$ docker inspect busybox:latest
Error: No such image or container: busybox:latest
[]
(flocker)jean-paul@dilaton:~$ 
```

Note that the container still reports it is using `busybox:latest` even after that tag ceases to exist locally.  Even worse, the tag can reappear (perhaps associated with a different image - for example, because you did a `docker pull` and the `latest` tag moved...) and inspect **still** claims it's running that tag... which now does exist but refers to the wrong image!  (And therefore possibly the wrong environment.)

I can't tell if these terrible cases are the ones we're actually encountering in the test suite because the test failures don't include enough information to actually understand what leads to them (yes, removing some image failed ... because it doesn't exist ... but who knows who created the image or who removed it later or why there's still a container using it).

They seem like issues pretty likely to matter in production use - at the very least it seems like this could lead to spurious application restarts (on top of new images!) .  Please file a follow-up issue for addressing them.